File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        sum, pre = 0, 0
        st = set()
        for i in nums:
            sum += i
            if sum % k in st:
                return True
            st.add(pre)
            pre = sum % k
        return False
        

File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {
        int sum = 0, pre = 0;
        unordered_set<int> set;
        for (int & i : nums) {
            sum += i;
            if (set.find(sum % k) != set.end()) {
                return true;
            }
            set.insert(pre);
            pre = sum % k;
        }
        return false;
    }
};